services/src/WebAPI/Controllers/User/UserQueryController.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Application.Models;
+using Application.Models.Pagination;
+using Application.Users.Queries.GetAllUsers;
+using Application.Users.Queries.GetLoggedUser;
+using Application.Users.Queries.GetUser;
+using Domain.Common.Pagination.Response;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using WebAPI.Common.Authorization;
+
+namespace WebAPI.Controllers.User;
+
+[ApiController]
+[Route("api/v1/users")]
+
+
+public class UserQueryController : ControllerBase
+{
+    private readonly IMediator _mediator;
+
+    public UserQueryController(IMediator mediator)
+    {
+        _mediator = mediator;
+        
+    }
+    
+    [HttpGet]
+    [AuthorizeAdmin]
+    public async Task<ActionResult<IEnumerable<UserDetailsDto>>> GetInfoAboutUser([FromQuery] string pageRequestDto)
+    {
+        var userInfo = await _mediator.Send(new GetUserQuery(pageRequestDto));
+        return Ok(userInfo);
+    }
+    
+    [HttpGet]
+    [AuthorizeAdmin]
+    [Route("AllUsers")]
+    public async Task<ActionResult<Page<UserDetailsDto>>> GetAllAsync([FromQuery] PageRequestDto pageRequestDto)
+    {
+        var result = await _mediator.Send(new GetAllUsersQuery(pageRequestDto));
+        return Ok(result);
+    }
+    
+    [HttpGet]
+    [AuthorizeAdmin]
```

Comment:
As i understand this endpoint is suppose to return info about logged user, so why to restrict it only for admin? Why regular user shouldn't have access to info about his own account?